src/features/lireFlow/hooks/useWordHighlighter.ts

```diff
@@ -1,119 +1,213 @@
+        }
 
-  const updateReadingState = useCallback(() => {
-    isReadingRef.current = isReading;
+        return hyphenated.slice(0, 1) + '-' + hyphenated.slice(1);
+      };
 
-    const calculateWordTime = async (word: string): Promise<number> => {
-      const hyphenatedText: string = await hyphenate(word, { hyphenChar: '-' });
-      const syllablesCount: number = hyphenatedText.split('-').length;
-      const variableNameWithAnMeaning =
-        (syllablesCount * averageSyllableTime(wordsPerMinuteRef.current)) /
-        speedRef.current;
-      return Math.round(variableNameWithAnMeaning);
-    };
+      hyphenated = fixHyphenationWordsStartWithVowel(hyphenated);
+    }
+
+    if (endWithTwoVowels.test(word)) {
+      const applyHiatusHyphenationRule = (hyphenated: string) => {
+        const lastIndex = word.length - 1;
+        const lastLetter = word[lastIndex];
+        const penulLetter = word[lastIndex - 1];
+        const isHiatus =
+          (strongVowels.test(penulLetter) && strongVowels.test(lastLetter)) ||
+          (strongVowels.test(penulLetter) && weakVowels.test(lastLetter));
+
+        if (isHiatus) {
+          return (
+            hyphenated.slice(0, lastIndex) + '-' + hyphenated.slice(lastIndex)
+          );
+        }
 
-    const highlightFlow = async (): Promise<void> => {
-      while (indexRef.current < elementIndexs.current.length) {
-        if (!isReadingRef.current) return;
+        return word;
+      };
+      hyphenated = applyHiatusHyphenationRule(hyphenated);
+    }
 
-        const element: string =
-          wordsAndSeparators[elementIndexs.current[indexRef.current]];
+    if (!hyphenated.includes('-') && word.length > 3) {
+      hyphenated = word.slice(0, 2) + '-' + word.slice(2);
+    }
 
-        let waitTime: number = 0;
+    return hyphenated;
+  };
 
-        if (isWord(element)) {
-          setCurrentIndex(elementIndexs.current[indexRef.current]);
-          waitTime = await calculateWordTime(element);
-        } else {
-          waitTime =
-            punctuationMarksTime.find(
-              (mark: { mark: string }) => mark.mark === element,
-            )?.time || 150;
-        }
+  const calculateWordTime = async (word: string): Promise<number> => {
+    const hyphenatedText: string = fixWordHyphenation(
+      word,
+      await hyphenate(word, { hyphenChar: '-' }),
+    );
+
+    const syllablesCount: number = hyphenatedText
+      .split('-')
+      .filter((syllable) => syllable.trim() !== '').length;
+
+    return Math.round(
+      (syllablesCount * averageSyllableTime(wordsPerMinuteRef.current)) /
+        speedRef.current,
+    );
+  };
+
+  const isNotActiveFlow = (flowId: number): boolean => {
+    return activeFlowIdRef.current !== flowId;
+  };
+
+  const highlightFlow = async (flowId: number): Promise<void> => {
+    while (indexRef.current < elementIndexs.current.length) {
+      if (!isReadingRef.current || isNotActiveFlow(flowId)) return;
```

Comment:
Try to put the if logic in a meaning variable 
```suggestion
      const meaningVariableName = !isReadingRef.current || isNotActiveFlow(flowId);
      
      if (meaningVariableName ) return;
```